src/contract.py

```diff
@@ -0,0 +1,136 @@
+from __future__ import annotations
+from abc import ABC, abstractmethod
+from enums import *
+import numpy as np
+
+
+class Contract(ABC):
+    _underlying: Stock
```

Comment:
Why do you use underscores before attributes?

Reply:
Those are access modifiers in python which indicate that those attributes are protected and should be handled that way by developers.

Reply:
Actually, I've just realized that I incorrectly declared these as class attributes. That need to be fixed.